mongorestore/metadata.go

```diff
@@ -288,6 +288,12 @@ func (restore *MongoRestore) createCollectionWithApplyOps(session *mongo.Client,
 }
 
 func createCollectionCommand(intent *intents.Intent, options bson.D) bson.D {
+	for i, elem := range options {
```

Comment:
This code doesn't seem to be checking the version of the server at all. Did we decide to disallow this on all versions even though it's not supported on 4.2 (or is it 4.0?)